IntelliTect.TestTools.WindowsTestWrapper/ExampleProjects/Notepad.AutomatedTests/Helpers/NotepadWindowHelpers.cs

```diff
@@ -15,16 +16,75 @@ public class NotepadWindowHelpers
 
         public bool CheckTextFileForExpectedText(string location, string fileName, string expectedText)
         {
-            Thread.Sleep( 100 );    //Wait just in case the file is being saved. If this ever fails, write a polling loop
-            return File.ReadAllText( Path.Combine( location, fileName ) ) == expectedText;
+            //Thread.Sleep( 100 );    //Wait just in case the file is being saved. If this ever fails, write a polling loop
+            //return File.ReadAllText( Path.Combine( location, fileName ) ) == expectedText;
+
+            bool result = false;
+            string combinedPath = Path.Combine(location, fileName);
+            int retryCount = 0;
+            int numberOfRetries = 10;
+
+            do
+            {
+                try
+                {
+                    if ( File.ReadAllText( combinedPath ) == expectedText )
+                    {
+                        result = true;
+                    }
+                }
+                //Catch just in case the file has not yet been saved
+                catch ( FileNotFoundException )
+                {
+                    if (retryCount == numberOfRetries)
+                    {
+                        Debug.WriteLine($"Could not read file after {numberOfRetries} attempts.");
+                        throw;
+                    }
+                }
+                retryCount++;
+                Thread.Sleep(100);
+            } while (retryCount < numberOfRetries);
+
+
+            return result;
```

Comment:
Nope. I kinda sped through this and forgot to review. Thank you for the catch (along with the ones below!) I'll make the changes on lunch.